src/server.cpp

```diff
@@ -2148,6 +2161,62 @@ void Server::sendAddNode(v3s16 p, MapNode n, u16 ignore_id,
 	}
 }
 
+void Server::sendMetadataChanged(const std::list<v3s16> &meta_updates, float far_d_nodes)
+{
+	float maxd = far_d_nodes * BS;
+	NodeMetadataList *meta_updates_list = new NodeMetadataList();
+	std::vector<u16> clients = m_clients.getClientIDs();
+
+	m_clients.lock();
+
+	for (u16 i : clients) {
+		RemoteClient *client = m_clients.lockedGetClientNoEx(i);
+		if (!client)
+			continue;
+
+		if (client->net_proto_version >= 37) {
+			ServerActiveObject *player = m_env->getActiveObject(i);
+
+			for (v3s16 pos : meta_updates) {
+				NodeMetadata *meta = m_env->getMap().getNodeMetadata(pos);
+
+				if (!meta)
+					continue;
+
+				if (player) {
+					// If player is far away, only set modified blocks not sent
+					v3f player_pos = player->getBasePosition();
+					if (player_pos.getDistanceFrom(intToFloat(pos, BS)) > maxd) {
+						client->SetBlockNotSent(getNodeBlockPos(pos));
+						continue;
+					}
+				}
+
+				// Add the change to send list
+				meta_updates_list->set(pos, meta);
+			}
+			// Send the meta changes
+			std::ostringstream os(std::ios::binary);
+			meta_updates_list->serialize(os, client->net_proto_version, false, true);
```

Comment:
@sfan5 3rd argument (`false`) sets `disk = false`, which prevents private metadata leaking.

Reply:
Yes, but my question is will it send metadata updates if private metadata changes (which the client would not see)?